### PR TITLE
shu/ranking pipeline

### PR DESCRIPTION
**Pull request by wintonzheng**
- rename candidate_id back to product_id for ranking piepline response schema
- remove query from ranking request

- [ ] Does this PR have impact on local development experience? If yes, make sure you have a plan and add the documentations to address issues that come with the change
- [ ] bump version
- [ ] make a release
- [ ] publish to pypi service

### CODE REVIEW

File: wyvern/components/ranking_pipeline.py [hunk 1]
```diff
@@ -67,13 +69,16 @@ class RankingResponse(BaseModel):
         events: the list of logged events
     """
 
-    ranked_candidates: List[ResponseCandidate]
+    ranked_products: List[ResponseCandidate]
```
Comment (suchintan) [inline wyvern/components/ranking_pipeline.py:72]: Just curious: Why is this `ranked_products`? Candidate is a generic term, but product is specific to a type of entity

File: wyvern/components/ranking_pipeline.py [hunk 2]
```diff
@@ -83,71 +88,70 @@ class RankingPipeline(
...
-            RankingRequest[WYVERN_ENTITY],
+            RANKING_REQUEST,
         ]()
         self.impression_logging_component = ImpressionEventLoggingComponent[
             WYVERN_ENTITY,
-            RankingRequest[WYVERN_ENTITY],
+            RANKING_REQUEST,
         ]()
+        if not business_logic:
+            business_logic = BusinessLogicPipeline[WYVERN_ENTITY, RANKING_REQUEST]()
```
Comment (suchintan) [inline wyvern/components/ranking_pipeline.py:113]: Should this be defaulted in the method signature?

File: wyvern/components/ranking_pipeline.py [hunk 3]
```diff
@@ -228,3 +229,6 @@ async def rank_candidates(
             business_logic_request,
         )
         return business_logic_response.adjusted_candidates
+
+    def bypass_ranking(self, request: RankingRequest[WYVERN_ENTITY]) -> bool:
```
Comment (suchintan) [inline wyvern/components/ranking_pipeline.py:233]: nit; I would add some documentation here

**Review by suchintan — APPROVED** [review]

File: wyvern/components/ranking_pipeline.py (hunk 1 above)
Comment (wintonzheng) [reply wyvern/components/ranking_pipeline.py:72]: Inventa has been using the `ranked_products` as the schema for all of the use cases. We don't want them to do a API migration on their server side any more.  i can also build pydantic "alias" for this field 🤔

File: wyvern/components/ranking_pipeline.py [hunk 4]
```diff
@@ -20,29 +24,27 @@
...
-from wyvern.wyvern_typing import WYVERN_ENTITY
+from wyvern.wyvern_typing import GENERALIZED_WYVERN_ENTITY, WYVERN_ENTITY
 
 
 class RankingRequest(
     BaseWyvernRequest,
     PaginationFields,
-    Generic[WYVERN_ENTITY],
+    CandidateSetEntity,
+    Generic[GENERALIZED_WYVERN_ENTITY],
```
[the remark this answers is not in the document]
Comment (wintonzheng) [reply wyvern/components/ranking_pipeline.py:38]: The generic here is indeed either `WYVERN_REQUEST or WYVERN_ENTITY`.

There's a plan to make WYVERN_REQUEST a child class of wyvern entity.

File: wyvern/components/ranking_pipeline.py (hunk 1 above)
Comment (suchintan) [reply wyvern/components/ranking_pipeline.py:72]: oh. Is it possible to decouple this object from the actual response object? Maybe we can re-name it at the very end or something like that.

They'll definitely want `ranked_products` or `ranked_brands` depending on the pipeline.

**Pull request closed without merging**